code/io/joy.h

```diff
@@ -214,8 +238,11 @@ namespace io
 			SCP_vector<coord2d> _ballValues; //!< The ball values
 			SCP_vector<HatPosition> _hatValues; //!< The current hat values
 
-			SCP_vector<int> _buttonDownTimestamp; //!< The timestamp since when the button is pressed, -1 if not pressed
-			SCP_vector<int> _buttonDownCount; //!< The number of times the button was pressed
+			SCP_vector<int> _buttonDownTimestamp;       //!< The timestamp since when the button is pressed, -1 if not pressed
+			SCP_vector<int> _buttonDownCount;           //!< The number of times the button was pressed
+			SCP_vector<int> _hatDownTimestamp;          //!< The timestamp when the hat activated, -1 if inactive (centered)
+			SCP_vector<SCP_vector<int>> _hatDownCount;  //!< The number of times each hat position has been hit [hat][hatpos]
```

Comment:
Could you put the various hat values in its own struct? That would make it clearer which values belong together.